vita3k/regmgr/src/regmgr.cpp

```diff
@@ -273,62 +278,117 @@ static void init_system_dreg(RegMgrState &regmgr, const std::wstring pref_path)
+    return category.empty() || name.empty();
 }
 
 static std::string fix_category(const std::string &category) {
     return category.back() == '/' ? category : category + '/';
 }
 
+static std::string set_default_value(RegMgrState &regmgr, const std::string &category, const std::string &name) {
+    if (category_or_name_is_empty(category, name))
+        return {};
+
+    const auto &reg = std::find_if(reg_template[category].begin(), reg_template[category].end(), [&](const auto &n) {
+        return n.name == name;
+    });
+
+    if (reg == reg_template[category].end()) {
+        LOG_ERROR("No default value found for {}{}", category, name);
+        return {};
+    }
+
+    auto &sys = regmgr.system_dreg[category][name];
+    sys.clear();
+    sys.resize(reg->size);
+    sys.assign(reg->value.begin(), reg->value.end());
+
+    LOG_INFO("Successfully set default value for {}{}", category, name);
+
+    save_system_dreg(regmgr);
+
+    return reg->value;
+}
+
 void get_bin_value(RegMgrState &regmgr, const std::string &category, const std::string &name, void *buf, uint32_t bufSize) {
+    if (category_or_name_is_empty(category, name))
+        return;
+
     std::lock_guard<std::mutex> lock(regmgr.mutex);
     const auto &sys = regmgr.system_dreg[fix_category(category)][name];
 
     memcpy(buf, sys.data(), bufSize);
 }
 
-void set_bin_value(RegMgrState &regmgr, const std::wstring &pref_path, const std::string &category, const std::string &name, const void *buf, const uint32_t bufSize) {
+void set_bin_value(RegMgrState &regmgr, const std::string &category, const std::string &name, const void *buf, const uint32_t bufSize) {
+    if (category_or_name_is_empty(category, name))
+        return;
+
     std::lock_guard<std::mutex> lock(regmgr.mutex);
 
     const char *data = reinterpret_cast<const char *>(buf);
     regmgr.system_dreg[fix_category(category)][name].assign(data, data + bufSize);
 
-    save_system_dreg(regmgr, pref_path);
+    save_system_dreg(regmgr);
 }
 
 int32_t get_int_value(RegMgrState &regmgr, const std::string &category, const std::string &name) {
+    if (category_or_name_is_empty(category, name))
+        return 0;
+
     std::lock_guard<std::mutex> lock(regmgr.mutex);
 
-    const auto &sys = regmgr.system_dreg[fix_category(category)][name];
-    const auto value_str = std::string(sys.begin(), sys.end());
+    const auto cat = fix_category(category);
+    const auto &sys = regmgr.system_dreg[cat][name];
+    auto value_str = std::string(sys.begin(), sys.end());
+    if (!std::all_of(value_str.begin(), value_str.end(), [](unsigned char c) { return std::isdigit(c) || (c == '\0'); })) {
+        LOG_ERROR("Invalid value for {}{}: {}, attempt using default value!", cat, name, value_str);
+        value_str = set_default_value(regmgr, cat, name);
+        if (value_str.empty())
+            return 0;
+    }
 
     return std::stoi(value_str);
```

Comment:
Use a try catch block around std::stoi looking for std::invalid_argument instead, that would be much cleaner.

Reply:
For here no, using try catch juste get succès or no, and here if first char of value is number that works, but is wrong...
Because if string is 0rue it pass and succes std::stoi with take just 0
Here my code check every char , for can reset any wrong int value set in file
For exemple: if vakue is 0ru it detect is not number and reset value by default